port/unix/omrsharedhelper.c

```diff
@@ -434,7 +434,7 @@ omr_unlinkControlFile(struct OMRPortLibrary* portLibrary, const char *controlFil
 				controlFileStatus->errorMsg = omrmem_allocate_memory(portLibrary, msgLen+1, OMR_GET_CALLSITE(), 
 OMRMEM_CATEGORY_PORT_LIBRARY);
 				if (NULL != controlFileStatus->errorMsg) {
-					strncpy(controlFileStatus->errorMsg, unlinkErrMsg, msgLen);
+					strcpy(controlFileStatus->errorMsg, unlinkErrMsg);
 					controlFileStatus->errorMsg[msgLen] = '\0';
```

Comment:
No need to set the null terminator anymore.

Reply:
couldn't this also have been resolved by just copying one extra byte (i.e. msgLen+1) ?

Reply:
Thank You for you comments and suggestions @youngar and @mstoodle.

Apart from using strcpy or doing msglen+1, there are other possible solutions to this issue. 

1. Using msgLen-1. [Link Here](https://us-cert.cisa.gov/bsi/articles/knowledge/coding-practices/strncpy-and-strncat) 
2. Replacing strncpy with memcpy.

I was able to build successfully on my local machine with strcpy, memcpy, msgLen+1 and msgLen-1 version. I think we can go ahead with msgLen+1 in strncpy (msgLen+1 for the null terminator too).

Reply:
+1 to @mstoodle's suggestion. Since you have to find the length of the string to make an allocation, you don't need to use `strcpy` or `strncpy`. Use `memcpy` to copy `msgLen + 1` characters, which will include the null terminator.

Reply:
Yeah, that's even better!

Reply:
I used another approach in #5503.